### ISSUE TITLE
[BUG] @JLayer changes Action behaviour.

### ISSUE THREAD

**Issue by gpucce** [Wikunia/Javis.jl#417]
**Describe the bug**

@JLayers changes an Action behaviour it shouldn't.

**To Reproduce**

1. Julia Version (i.e. output of `julia -v`): 1.6.2

2. Operating system (Mac, Linux, Windows): Windows 10

3. Javis version (i.e output of `] status Javis` in the REPL) 0.6.2

4. Minimum working code example that led to bug:

```julia
using Javis

nolayer_video = Video(500, 500)

function ground(args...)
    background("black") # canvas background
    sethue("white") # pen color
end

Background(1:140, ground)

circ = Object(JCircle(O, 20, action=:fill, color="white"))

act!(circ, Action(1:70, anim_translate(Point(150, 150))))
act!(circ, Action(70:140, anim_translate(Point(-150, -150))))

render(nolayer_video, pathname="mwe1.gif")


layer_video = Video(500, 500)

Background(1:140, ground)

l1 = @JLayer 1:140 begin
    Object(JCircle(O, 20, action=:fill, color="white"))
end

act!(l1, Action(1:70, anim_translate(Point(150, 150))))
act!(l1, Action(70:140, anim_translate(Point(-150, -150))))

render(layer_video, pathname="mwe2.gif")
```

mwe1.gif:
![mwe1](https://user-images.githubusercontent.com/32967787/133002161-8c97ca95-0c64-4a4d-a461-0b57bb3965d7.gif)

mwe2.gif:
![mwe2](https://user-images.githubusercontent.com/32967787/133002163-9f52e65e-bb1e-48ee-af46-c5d7e066037d.gif)

**Comment by gpucce** [comment]
I think this happens because the second action uses the starting position of the Layer instead of the current one. If you can confirm this makes sense or point me a bit where to look. I can try to work on the issue myself!!

**Comment by TheCedarPrince** [comment]
Hey @gpucce - something is definitely going wrong here. Thanks for providing the Minimal working examples. As you see in example 1, it works fine for standard objects. Something is going wrong with layers as `keep` should be "keeping" the last position of where the layer was moved to. @Wikunia and @Sov-trotter , I tried explicitly setting `keep = true` to see if the behavior continued and it does. I am thinking it has something to do with what the layer returns. However, I do not know where to look just yet as it does not throw an error. Any thoughts?

**Comment by gpucce** [comment]
Hi thanks for replying, reading Can you help me understand why it works without layer

**Comment by Wikunia** [comment]
@Sov-trotter as the anim translate should be fixed now regarding issue #369 and as it works for objects as expected now (hopefully as it's written in the docs 😊 ) I would say this is an error regarding layers only. I can have a look at this in a few minutes

**Comment by gpucce** [comment]
I'm trying to work on this with advice from @Wikunia.
